io.github.sgsaenger.vipster.yml

```diff
@@ -0,0 +1,69 @@
+id: io.github.sgsaenger.vipster
+runtime: org.kde.Platform
+runtime-version: '6.8'
+sdk: org.kde.Sdk
+command: vipster
+modules:
+  - name: nlohmann_json
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DJSON_BuildTests=OFF
+    sources:
+      - type: git
+        url: https://github.com/nlohmann/json.git
+        tag: v3.11.2
+        commit: 0ca0fe433eb70cea0d5761079c0c5b47b736565b
+  - name: fmt
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DFMT_DOC=OFF
+      - -DFMT_TEST=OFF
+      - -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE
+    sources:
+      - type: git
+        url: https://github.com/fmtlib/fmt
+        tag: 11.0.2
+        commit: 0c9fce2ffefecfdce794e1859584e25877b7b592
+  - name: pybind11
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DPYBIND11_TEST=OFF
+    sources:
+      - type: git
+        url: https://github.com/pybind/pybind11
+        tag: v2.13.6
+        commit: a2e59f0e7065404b44dfe92a28aca47ba1378dc4
+  - name: cli11
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCLI11_BUILD_TESTS=OFF
+      - -DCLI11_BUILD_EXAMPLES=OFF
+    sources:
+      - type: git
+        url: https://github.com/CLIUtils/CLI11
+        tag: v2.2.0
+        commit: b9be5b9444772324459989177108a6a65b8b2769
+  - name: vipster
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DBUILD_TESTING=OFF
+      - -DVIPSTER_DOWNLOAD_DEPENDENCIES=OFF
+      - -Dtinyexpr_SOURCE_DIR=/run/build/vipster/tinyexpr
+    sources:
+      - type: git
+        url: https://github.com/sgsaenger/vipster.git
+        commit: 044fe23e455a132c451ccf43f7a21450c268e6a0
+      - type: git
+        url: https://github.com/codeplea/tinyexpr.git
+        commit: 373d2d3d5dc3344e3e2cb826717404064562a058
+        dest: tinyexpr
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=fallback-x11
+  # Wayland access
+  - --socket=wayland
+  # GPU acceleration if needed
+  - --device=dri
+  # Needs to save files locally
+  - --filesystem=home
```

Comment:
doesn't this use portal? Qt support it be default.

Reply:
Not sure how portal works, without `--filesystem=home` it just presents an empty 'home' dir on my local machine.
Further, Vipster can be used for batch-conversion of files as a commandline utility without using any Qt functionality.
Any recommendations?

Reply:
>  without --filesystem=home it just presents an empty 'home' dir on my local machine.

then it doesn't use the portal.

Reply:
After reading up some more on portal, i'd say relying fully on portal is not possible at the moment, as the command line functionality would break.
I need to investigate either using Qt for command line handling or accessing the portal API directly.
Will put this on the roadmap for V2.

Would there be any danger in using `filesystem=home` for the current version, in the sense that Qt might confuse things with this option and portal?

Reply:
Not any danger but using portals is better for sandboxing as it selectively gives access to files/folders upon user interaction while static permissions expose everything in home always.